src/primitive/measure.jl

```diff
@@ -1,31 +1,32 @@
-using YaoBase, YaoArrayRegister
+using YaoBase, YaoArrayRegister, Random
 export Measure, AllLocs, ComputationalBasis
 
 """
-    Measure{N, K, OT} <: PrimitiveBlock{N}
-    Measure(n::Int; operator=ComputationalBasis(), locs=1:n, collapseto=nothing, remove=false)
+    Measure{N, K, OT, RNG} <: PrimitiveBlock{N}
+    Measure(n::Int; rng=Random.GLOBAL_RNG, operator=ComputationalBasis(), locs=1:n, collapseto=nothing, remove=false)
 
 Measure operator.
 """
-mutable struct Measure{N, K, OT} <: PrimitiveBlock{N}
+mutable struct Measure{N, K, OT, RNG} <: PrimitiveBlock{N}
+    rng::RNG
     operator::OT
     locations::Union{NTuple{K, Int}, AllLocs}
     collapseto::Union{Int, Nothing}
     remove::Bool
     results::Vector{Int}
-    function Measure{N, K, OT}(operator, locations, collapseto, remove) where {N, K, OT}
+    function Measure{N, K, OT, RNG}(rng::RNG, operator, locations, collapseto, remove) where {RNG, N, K, OT}
         locations isa AllLocs || @assert_locs_safe N locations
         if collapseto !== nothing && remove == true
             error("invalid keyword combination, expect collapseto or remove, got (collapseto=$collapseto, remove=true)")
         end
-        new{N, K, OT}(operator, locations, collapseto, remove)
+        new{N, K, OT, RNG}(rng, operator, locations, collapseto, remove)
     end
 end
 
 @interface nqubits_measured(::Measure{N, K}) where {N, K} = K
 
 """
-    Measure(n::Int; operator=ComputationalBasis(), locs=AllLocs(), collapseto=nothing, remove=false)
+    Measure(n::Int; rng=Random.GLOBAL_RNG, operator=ComputationalBasis(), locs=AllLocs(), collapseto=nothing, remove=false)
```

Comment:
The position looks inconsistent with the rest of them. Can we have `rng` as the first optional argument? So this is consistent with other functions instead of a kwarg

Reply:
this is in purpose.

Reply:
This function is different

Reply:
Explain why?

Reply:
We still have the original interface. And this is consistent with all other Julia interface which is more intuitive. Explain what's the benefits, or we shouldn't just create our own interface. Especially this is widely used.

Reply:
For blocks, we have a different convention. This is consistent with the original one.